### DISCUSSION TITLE
We're seeing an 80-fold increase in function execution time since 31 May - anyone else?

### DISCUSSION THREAD

**Discussion by jbothma** [vercel/community#3062] [Feature Question]
### Question

Since 31 May we went from about 1GB-Hr/day to about 80GB-Hr/day

The only changes that day were

- Bump next.js 13.4.1 to 13.4.4
- Bump bootstrap from 5.2.3 to 5.3.0
- some content on a statically-built page

![Screenshot_2023-06-29_17-28-53](https://github.com/vercel/community/assets/235801/3361ab13-5a7c-4967-bbcd-4a443481e872)

The chart showing execution time by function doesn't show the same order of magnitude increase. And when I download the data and sum it all up, it's also roughly the same total.

![Screenshot_2023-06-22_08-23-41-masked](https://github.com/vercel/community/assets/235801/8d4bf570-fb7d-41d5-873d-3af98c097bbd)

Has anyone else seen a similar change? Any ideas? It's not even clear to me that the monitoring add-on would give me more information and is worth paying for since the function summary which should already show the big functions isn't shedding light.

**Comment by amyegan** [comment]
Sorry you're seeing a big spike in usage. If you share a deployment id or url, I can look into the logs for more details.

**Reply by jbothma** [reply]
Hi! Thank you!
It's https://vercel.com/opensanctions/site

**Reply by jbothma** [reply]
More specifically, https://www.opensanctions.org/

**Reply by amyegan** [reply]
Thanks for sharing that. We're looking into it now 🙇

**Comment by jbothma** [comment]
I forgot to mention we downgraded next.js back to 13.4.1 for a number about 5 days and didn't see the spend drop.

**Comment by amyegan** [comment]
It looks like the site uses Streaming with Serverless Functions, which became generally available about a month ago. The timing of the increase in execution time coincides with the start of billing for streaming execution time with the stable release. 

Next.js App Router has support for [Streaming and Suspense](https://nextjs.org/docs/app/building-your-application/data-fetching#streaming-and-suspense). While these features are beneficial in most cases, long-running streams can lead to higher execution time. You can find more info in the [Streaming with Suspense](https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming#streaming-with-suspense) and [Streaming in Route Handlers](https://nextjs.org/docs/app/building-your-application/routing/router-handlers#streaming) docs.

**Reply by jbothma** [reply]
Thank you! I'll look at that.

Can you tell me more about how I might have figured this out from the usage data? It's been frustrating that we didn't really see hints about which functions are consuming so much more resources. While a few of the top functions by consumption did consume more resources than the previous top functions, the number of executions didn't show much of an increase. So there aren't really hints to follow in the usage metrics.

**Reply by amyegan** [reply]
We're looking into why the usage report doesn't adequately reflect the breakdown of usage. I apologize for the lack of clarity in my previous post. I only meant to communicate what I learned about the general cause of increased execution time. 

The team is working to identify an appropriate solution to ensure all customers have the level of detail necessary to debug routes and control usage. If you have any billing issues related to this problem, please do not hesitate to [submit a billing support case](https://vercel.com/help#issues).

**Reply by jbothma** [reply]
Would activation of billing for streaming execution time cause the increase in logged execution time? Because we're not just seeing an increase in billing, we're seeing an increase in recorded GB-Hrs. It would have made sense if we could see the execution time whether we're billed for it or not.

**Comment by pudo** [comment]
So just to clarify: because we have used `loading.tsx` on a few of our screens, there's now an unannounced 80x cost increase on hosting the web page on Vercel? That's a complete breach of trust.

**Reply by amyegan** [reply]
I apologize for my lack of clarity. We are still looking into the sudden unexpected change in usage. My earlier post was only meant to communicate what I know so far about potential causes of higher execution time.

We are not finished investigating the situation and you'll continue to receive updates here as more information becomes available.  🙇

**Comment by jbothma** [comment]
We switched to the App router in December, and we didn't see a significant change (up or own) in execution time then. (We were already heading upwards in November.) Most of our traffic is to dynamic routes which were server-side rendered before the App router.

I'm not sure I understand why Suspend/Streaming would change the cost - the function is spending the same amount of time executing, effectively, if our upstream API latency hasn't changed, right?

It looks like there was massive drop in recorded execution time in February and then it came back really really high in June. It's a bit difficult to know exactly why it's so large now and whether that is correct.

We're turning off pre-fetching of dynamic routes for now - it's possible that part of the issue is that RSC requests have increased. We're not seeing this increase on the invocations chart, so at the moment we're still unsure what's going on really. Thanks for continuing the investigation - we're looking forward to your findings.

![Screenshot_2023-07-03_09-21-57](https://github.com/vercel/community/assets/235801/4b690a00-dc5e-4081-a31a-37f2e993dbeb)
